ui/desktop/src/components/ReportFailureModal.tsx

```diff
@@ -0,0 +1,264 @@
+import { useState, useEffect, useCallback } from 'react';
+import { Button } from './ui/button';
+import { Card, CardContent } from './ui/card';
+import { AlertCircle, CheckCircle, Loader2 } from 'lucide-react';
+import { Dialog, DialogContent, DialogHeader, DialogTitle } from './ui/dialog';
+import { getExtensions, providers } from '../api/sdk.gen';
+
+interface SystemInfo {
+  gooseVersion: string;
+  osVersion: string;
+  platform: string;
+  architecture: string;
+  providerType?: string;
+  extensionCount: number;
+}
+
+interface ReportFailureModalProps {
+  isOpen: boolean;
+  onClose: () => void;
+}
+
+export default function ReportFailureModal({ isOpen, onClose }: ReportFailureModalProps) {
+  const [description, setDescription] = useState('');
+  const [systemInfo, setSystemInfo] = useState<SystemInfo | null>(null);
+  const [isSubmitting, setIsSubmitting] = useState(false);
+  const [submitStatus, setSubmitStatus] = useState<'idle' | 'success' | 'error'>('idle');
+
+  const getExtensionCount = async (): Promise<number> => {
+    try {
+      const response = await getExtensions({});
+
+      if (response.data && Array.isArray(response.data)) {
+        return response.data.length;
+      }
+    } catch (error) {
+      console.error('Failed to get extension count:', error);
+    }
+    return 0;
+  };
+
+  const collectSystemInfo = useCallback(async () => {
+    try {
+      const info: SystemInfo = {
+        gooseVersion: String(
+          window.appConfig?.get('GOOSE_VERSION') ||
+            window.appConfig?.get('version') ||
+            'Development'
+        ),
+        osVersion: navigator.userAgent,
+        platform: window.electron?.platform || 'Unknown',
+        architecture: window.electron?.arch || 'Unknown',
+        extensionCount: await getExtensionCount(),
+      };
+
+      try {
+        const providersResponse = await providers();
+        if (providersResponse.data && Array.isArray(providersResponse.data)) {
+          const providersList = providersResponse.data;
+          if (providersList.length > 0) {
+            info.providerType = providersList[0].name || 'Unknown';
```

Comment:
this isn't the right one to use. You can use `useModelAndProvider` to get the current model and provider